xunit-autolabeler-v2/ast_parser/core/cli_yaml_errors.py

```diff
@@ -0,0 +1,101 @@
+# Copyright 2020 Google LLC. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+This file contains string descriptions of YAML validation errors. These
+errors are represented as classes to a) make message modification easier
+and b) enable direct isinstance() checks.
+"""
+
+from typing import NamedTuple
+
+
+class InvalidAttributeViolation(NamedTuple):
```

Comment:
Dataclasses (https://docs.python.org/3/library/dataclasses.html) are the preferred way of doing this in Python 3.7. Sorry I previously suggested NamedTuples, we just migrated our codebase to Python 3.7 and I'm learning the new standards :)

Reply:
Done.